### PR TITLE
Remove unnecessary if statement in pv_controller.go

### PR DESCRIPTION
**Pull request by mysunshine92**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
 /kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mysunshine92** [comment]
cc @mattjmcnaughton

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 1]
```diff
@@ -474,30 +462,24 @@ func (ctrl *PersistentVolumeController) syncBoundClaim(claim *v1.PersistentVolum
 			// volume yet. We can't distinguish these cases.
 			// Bind the volume again and set all states to Bound.
 			klog.V(4).Infof("synchronizing bound PersistentVolumeClaim[%s]: volume is unbound, fixing", claimToClaimKey(claim))
-			if err = ctrl.bind(volume, claim); err != nil {
-				// Objects not saved, next syncPV or syncClaim will try again
-				return err
-			}
-			return nil
+			// Objects not saved, next syncPV or syncClaim will try again
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: My concern is that the lack of if statement now makes it look as if this `// Objects not saved...` comment is always true. It is only the case if the `ctrl.bind` command fails (i.e. returns an error).

Could you update the comment to indicate that?

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 2]
```diff
@@ -474,30 +462,24 @@ func (ctrl *PersistentVolumeController) syncBoundClaim(claim *v1.PersistentVolum
...
 			// NOTE: syncPV can handle this so it can be left out.
 			// NOTE: bind() call here will do nothing in most cases as
 			// everything should be already set.
 			klog.V(4).Infof("synchronizing bound PersistentVolumeClaim[%s]: claim is already correctly bound", claimToClaimKey(claim))
-			if err = ctrl.bind(volume, claim); err != nil {
-				// Objects not saved, next syncPV or syncClaim will try again
-				return err
-			}
-			return nil
+			// Objects not saved, next syncPV or syncClaim will try again
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Same as above

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 3]
```diff
@@ -513,24 +495,20 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
...
 		klog.V(4).Infof("synchronizing PersistentVolume[%s]: volume is unused", volume.Name)
-		if _, err := ctrl.updateVolumePhase(volume, v1.VolumeAvailable, ""); err != nil {
-			// Nothing was saved; we will fall back into the same
-			// condition in the next call to this method
-			return err
-		}
-		return nil
+		// Nothing was saved; we will fall back into the same
+		// condition in the next call to this method
+		_, err := ctrl.updateVolumePhase(volume, v1.VolumeAvailable, "")
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Samve as above - it appears that it's always the case that nothing was saved.

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 4]
```diff
@@ -513,24 +495,20 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
...
 			// The PV is reserved for a PVC; that PVC has not yet been
 			// bound to this PV; the PVC sync will handle it.
 			klog.V(4).Infof("synchronizing PersistentVolume[%s]: volume is pre-bound to claim %s", volume.Name, claimrefToClaimKey(volume.Spec.ClaimRef))
-			if _, err := ctrl.updateVolumePhase(volume, v1.VolumeAvailable, ""); err != nil {
-				// Nothing was saved; we will fall back into the same
-				// condition in the next call to this method
-				return err
-			}
-			return nil
+			// Nothing was saved; we will fall back into the same
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Same as above.

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 5]
```diff
@@ -601,12 +579,10 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
 				}
 			}
 
-			if err = ctrl.reclaimVolume(volume); err != nil {
-				// Release failed, we will fall back into the same condition
-				// in the next call to this method
-				return err
-			}
-			return nil
+			// Release failed, we will fall back into the same condition
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Same as above.

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 6]
```diff
@@ -638,12 +614,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
...
 			// Volume is bound to a claim properly, update status if necessary
 			klog.V(4).Infof("synchronizing PersistentVolume[%s]: all is bound", volume.Name)
-			if _, err = ctrl.updateVolumePhase(volume, v1.VolumeBound, ""); err != nil {
-				// Nothing was saved; we will fall back into the same
-				// condition in the next call to this method
-				return err
-			}
-			return nil
+
+			// Nothing was saved; we will fall back into the same
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Same as above.

File: pkg/controller/volume/persistentvolume/pv_controller.go [hunk 7]
```diff
@@ -661,12 +636,11 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
...
 					}
 				}
-				if err = ctrl.reclaimVolume(volume); err != nil {
-					// Deletion failed, we will fall back into the same condition
-					// in the next call to this method
-					return err
-				}
-				return nil
+
+				// Deletion failed, we will fall back into the same condition
```
Comment (mattjmcnaughton) [inline pkg/controller/volume/persistentvolume/pv_controller.go]: Same as above.

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
/ok-to-test

Thanks for this change :)

Overall, the code looks good to me. My one concern is that I think some of the existing comments are quite confusing now. Mind updating them? I included some ideas in my review.

**Pull request closed without merging**